File: 프로그래머스/lv2/42746. 가장 큰 수/가장 큰 수.py
def solution(numbers):
    if sum(numbers) == 0:
        return "0"
    
    li = []
    for i in range(len(numbers)):
        a = (str(numbers[i]) * 4)[:4]
        b = len(str(numbers[i]))
        li.append((a,b))
        
    li.sort(reverse=True) 
    answer = ''
    for a, b in li:
        answer += a[:b]
    return answer

File: 프로그래머스/2/42746. 가장 큰 수/가장 큰 수.java
import java.util.*;

class Solution {
    public String solution(int[] numbers) {
       
        String[] strNumbers = new String[numbers.length];
        
        for (int i = 0; i < numbers.length; i++) {
            strNumbers[i] = String.valueOf(numbers[i]);
        }

        Arrays.sort(strNumbers, (o1, o2) -> (o2 + o1).compareTo(o1 + o2));
       
        if (strNumbers[0].equals("0")) {
            return "0";
        }

        StringBuilder result = new StringBuilder();
        for (String num : strNumbers) {
            result.append(num);
        }

        return result.toString();
    }
}
